pkg/standalone-utils/test/ProtocolFeeSplitter.test.ts

```diff
@@ -0,0 +1,229 @@
+
+    const request: JoinPoolRequest = {
+      assets: tokens.addresses,
+      maxAmountsIn: initialBalances,
+      userData: WeightedPoolEncoder.joinInit(initialBalances),
+      fromInternalBalance: false,
+    };
+
+    await vault.instance
+      .connect(liquidityProvider)
+      .joinPool(poolId, liquidityProvider.address, liquidityProvider.address, request);
+  });
+
+  sharedBeforeEach('deploy ProtocolFeeSplitter, ProtocolFeesWithdrawer & grant permissions', async () => {
+    protocolFeesWithdrawer = await deploy('ProtocolFeesWithdrawer', {
+      args: [vault.address, []],
+    });
+
+    protocolFeeSplitter = await deploy('ProtocolFeeSplitter', {
+      args: [protocolFeesWithdrawer.address, treasury.address],
+    });
+
+    const setRevenueSharingFeeRole = await actionId(protocolFeeSplitter, 'setRevenueSharingFeePercentage');
+    await vault.grantPermissionsGlobally([setRevenueSharingFeeRole], admin);
+
+    const setDefaultRevenueSharingFeePercentageRole = await actionId(
+      protocolFeeSplitter,
+      'setDefaultRevenueSharingFeePercentage'
+    );
+    await vault.grantPermissionsGlobally([setDefaultRevenueSharingFeePercentageRole], admin);
+
+    // Allow withdrawer to pull from collector
+    const withdrawCollectedFeesRole = await actionId(protocolFeesCollector, 'withdrawCollectedFees');
+    await vault.grantPermissionsGlobally([withdrawCollectedFeesRole], protocolFeesWithdrawer);
+
+    // Allow fee splitter to pull from withdrawer
+    const withdrawCollectedFeesWithdrawerRole = await actionId(protocolFeesWithdrawer, 'withdrawCollectedFees');
+    await vault.grantPermissionsGlobally([withdrawCollectedFeesWithdrawerRole], protocolFeeSplitter);
+
+    const setTreasuryRole = await actionId(protocolFeeSplitter, 'setTreasury');
+    await vault.grantPermissionsGlobally([setTreasuryRole], admin);
+  });
+
+  async function createPool(poolOwnerAddress: string): Promise<Contract> {
+    const receipt = await (
+      await factory.create(
+        NAME,
+        SYMBOL,
+        tokens.addresses,
+        WEIGHTS,
+        assetManagers,
+        POOL_SWAP_FEE_PERCENTAGE,
+        poolOwnerAddress
+      )
+    ).wait();
+
+    const event = expectEvent.inReceipt(receipt, 'PoolCreated');
+    return deployedAt('v2-pool-weighted/WeightedPool', event.args.pool);
+  }
+
+  describe('constructor', () => {
+    it('sets the protocolFeesWithdrawer', async () => {
+      expect(await protocolFeeSplitter.protocolFeesWithdrawer()).to.be.eq(protocolFeesWithdrawer.address);
+    });
+    it('sets the treasury', async () => {
+      expect(await protocolFeeSplitter.treasury()).to.be.eq(treasury.address);
+    });
+  });
+
+  it('changes the treasury', async () => {
+    const receipt = await (await protocolFeeSplitter.connect(admin).setTreasury(newTreasury.address)).wait();
+    expectEvent.inReceipt(receipt, 'TreasuryChanged', { newTreasury: newTreasury.address });
+  });
```

Comment:
There should be a test where an account with no permissions fails to change the treasury